dsgrid/config/project_config.py

```diff
@@ -613,6 +613,48 @@ def _iter_datasets_by_status(self, status: DatasetRegistryStatus) -> InputDatase
             if dataset.status == status:
                 yield dataset
 
+    @track_timing(timer_stats_collector)
+    def make_dimension_association_table(self, data_source=None):
+        """Return a table with associations across all dimensions.
+
+        Parameters
+        ----------
+        data_source : DimensionType, optional
+            Filter table by data_source, by default None.
+
+        Returns
+        -------
+        pyspark.sql.DataFrame
+
+        """
+        ds = DimensionType.DATA_SOURCE.value
+        table = self.dimension_associations.table
+        table_columns = set()
+        # table is None when the project doesn't define any dimension associations.
+        if table is not None:
+            if ds in table.columns and data_source is not None:
+                table = table.filter(f"{ds} = '{data_source}'").drop(ds)
+            table_columns.update(table.columns)
+
+        exclude = set((DimensionType.TIME, DimensionType.DATA_SOURCE))
```

Comment:
if data_source is None, we should not exclude `data_source` right?

Reply:
Yeah, good catch. In the location where I copied this code, data_source was always set.